### ISSUE TITLE
Support axes argument in transpose

### ISSUE THREAD

**Issue by ehsantn** [numba/numba#2745]
One often needs to convert images (stored in ndarrays) between formats like HWC (height/width/channel) and CHW, when working with deep learning frameworks like TensorFlow and PyTorch. This requires permutations of axes which transpose provides with the optional `axes` argument.

**Comment by ehsantn** [comment]
@agayev

**Comment by ehsantn** [comment]
The output array of `transpose` is possibly neither `C` layout nor `F` layout if we return a view. Can Numba handle this? @sklam @stuartarchibald 
We might need to copy the data. @agayev

**Comment by sklam** [comment]
@ehsantn, Yes, numba has a 'A' for "any" layout for arbitrarily strided array.

**Comment by ehsantn** [comment]
@sklam Ok, thanks. I thought 'A' layout has to be resolved to 'C' or 'F' after type inference.

We assume 'C' layout in ParallelAccelerator which needs to be fixed.
